pkg/resource/broker/sdk.go

```diff
@@ -565,14 +581,18 @@ func (rm *resourceManager) sdkUpdate(
 		return nil, err
 	}
 
-	resp, respErr := rm.sdkapi.UpdateBrokerWithContext(ctx, input)
-	rm.metrics.RecordAPICall("UPDATE", "UpdateBroker", respErr)
-	if respErr != nil {
-		return nil, respErr
+	var resp *svcsdkapi.UpdateBrokerResponse
+	resp, err = rm.sdkapi.UpdateBrokerWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateBroker", err)
+	if err != nil {
+		return nil, err
 	}
 	// Merge in the information we read from the API call above to the copy of
 	// the original Kubernetes object we passed to the function
 	ko := desired.ko.DeepCopy()
+	// Copy status from latest observed state
+	latestKOStatus := latest.ko.DeepCopy().Status
+	ko.Status = latestKOStatus
```

Comment:
I'm think that maybe this should Go to the code-generator

Reply:
Maybe we should change this:

```go
ko := desired.ko.DeepCopy()
```

to this in the code-generator you mean?

```go
ko := latest.ko.DeepCopy()
```

Since `latest` will already be merged with `desired` from the ReadOne call...

Reply:
>Since latest will already be merged with desired from the ReadOne call...

Yep ++

Reply:
I just tested with a `ko := latest.ko.DeepCopy()` and it's working properly - i'll send a PR changing code-gen